guardian/frontend
# Prep newsletter embeds for new designs

## What does this change?
New designs are being introduced for newsletter embeds and a good amount
of health work is required in order to both maintain the general health
of the codebase and to allow integration of new designs with minimal
reliance on Editorial changes.

Here I've introduced a number of important changes. Firstly, the code
handling the rendering of the footer embeds is now separate from the
code for in article embeds. This should mean that it's easier to
implement specific design changes for each embed type without having
increasing levels of business logic tied into the rendering code.

Secondly, the Identity model has been updated to include an `EmailEmbed`
field which contains the copy that will be displayed for each in-article
embed. As a result, there is no need to pass this information to the
iframe as attributes. In addition, there is an endpoint per newsletter
for the success state. This allows specific copy for any one newsletter
following submission.

### How to test
~If https://github.com/guardian/identity/pull/1827 has not yet been merged and released, pull that PR and then use `publishLocal` to update the Identity model being used locally. Without this, it won't work.~ 

The new Identity model has been released and so `publishLocal` is not necessary.

~In addition, to really test how we want/expect this to work,  comment out/delete the `email.js` file. If this is running it will affect how the frontend embeds display. We want to move away from using this because it doesn't run on apps or DCR, making maintaining and using it both a liability and waste of time in the long term. In an ideal world this PR would make that entire module redundant, but that feels unlikely. So for now, just comment it out and let's see how it goes.~ 

`email.js` removed as part of this PR.

### Remaining:
- Replicate the `success`/`error` state tracking that is currently done by Google Analytics in `email.js`, using Ophan instead. Sign ups can be tracked using the work from https://github.com/guardian/frontend/pull/22806


## Does this change need to be reproduced in dotcom-rendering ?


This work should ideally improve DCR automatically by removing the reliance on frontend clientside code to control how the embeds render.

## Screenshots

## What is the value of this and can you measure success?

Easier to maintain and update newsletters. Able to implement the new designs for OKR.

## Checklist

### Does this affect other platforms?

- [ ] AMP 

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?




- [ ] Yes (please give details)

### Does this change break ad-free?






- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?




- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist




### Tested
